app/repositories/CacheRepository.scala

```diff
@@ -151,6 +153,13 @@ class CacheRepository @Inject() (
       totalMatchingDocuments
     )
   }
+
+  def existsLRN(lrn: String, eoriNumber: String): Future[Boolean] = { // TODO: CTCP-3469 Will want to filter for non submitted declarations (e.g drafts) can do this off the back of isSubmitted flag in userAnswers - if one is found set a flag potentially to say it is being used elsewhere
+    val lrnFilter     = Filters.eq("lrn", lrn)
+    val eoriFilter    = Filters.eq("eoriNumber", eoriNumber)
+    val primaryFilter = Filters.and(eoriFilter, lrnFilter)
```

Comment:
Interesting...Do we want to restrict this check to the single EORI? I would have thought the LRN check was a global search?

Reply:
Yeah that's a good point. If the draft exists anywhere then the local reference number gets used, that will want to be updated in the future regardless of the EORI number